Makefile

```diff
@@ -112,10 +114,11 @@ run-test: build
 	PID=$$!; \
 	echo "skiperator PID: $$PID"; \
 	echo "Log redirected to file: $$LOG_FILE"; \
-	if [ -z "$(TEST_DIR)" ]; then \
-		$(MAKE) test; \
-	else \
-		$(MAKE) test-single dir=$(TEST_DIR); \
-	fi; \
-	echo "Stopping skiperator (PID $$PID)..."; \
-	kill $$PID
+	( \
+		if [ -z "$(TEST_DIR)" ]; then \
+			$(MAKE) test; \
+		else \
+			$(MAKE) test-single dir=$(TEST_DIR); \
+		fi; \
+	) && \
+	(kill $$PID && echo "Stopping skiperator (PID $$PID)...")  || (echo "Test or skiperator failed" && exit 1)
```

Comment:
```shell
(echo "Stopping skiperator (PID $$PID)..." && kill $$PID)
```
So we're noticing which PID is about to be killed

Reply:
Changed the order, also added kill when the test fail